storages/backends/s3boto3.py

```diff
@@ -440,6 +440,10 @@ def _save(self, name, content):
         name = self._normalize_name(cleaned_name)
         params = self._get_write_parameters(name, content)
 
+        # wrap content so read() always returns bytes. This is required for passing it
+        # to obj.upload_fileobj() or self._compress_content()
+        content = ReadBytesWrapper(content)
```

Comment:
This is the core part of the fix. This approach has the benefits of not making any extra read() or seek() calls to the file object.